assembl/alembic/versions/0888e0f1a92d_timeline.py

```diff
@@ -0,0 +1,102 @@
+"""timeline
+
+Revision ID: 0888e0f1a92d
+Revises: 498e7af689d2
+Create Date: 2017-02-27 12:12:45.625018
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '0888e0f1a92d'
+down_revision = '498e7af689d2'
+
+from alembic import context, op
+import sqlalchemy as sa
+import transaction
+
+
+from assembl.lib import config
+from assembl.lib.sqla import mark_changed
+
+def upgrade(pyramid_env):
+    with context.begin_transaction():
+        op.add_column(
+            'timeline_event',
+            sa.Column('title_id', sa.Integer(),
+                      sa.ForeignKey('langstring.id')))
+        op.add_column(
+            'timeline_event',
+            sa.Column('description_id', sa.Integer(),
+                      sa.ForeignKey('langstring.id')))
+        op.add_column(
+            'timeline_event',
+            sa.Column('image_url', sa.String()))
+
+    # Do stuff with the app's models here.
+    from assembl import models as m
+    db = m.get_session_maker()()
+    with transaction.manager:
+        data = list(db.execute(
+            "SELECT discussion_id, id, title, description FROM timeline_event"))
+        if data:
+            discussion_ids = {d[0] for d in data}
+            discussions = db.query(m.Discussion).filter(m.Discussion.id.in_(discussion_ids)).all()
+            locales = {d.id: d.main_locale for d in discussions}
+            for d_id, ev_id, title, description in data:
+                title = m.LangString.create(title, locales[d_id])
+                description = m.LangString.create(description, locales[d_id])
+                db.add(title)
+                db.add(description)
+                db.flush()
+                db.execute(
+                    'UPDATE timeline_event SET title_id=%d, description_id=%d WHERE id = %d' % (
+                    title.id, description.id, ev_id))
+            mark_changed()
```

Comment:
Gotcha, good reminder. There are two transactions to be aware of, the zope one (for pyramid) and the one for alembic...